yt/data_objects/octree_subset.py

```diff
@@ -186,8 +186,8 @@ def deposit(self, positions, fields = None, method = None,
         op.initialize()
         mylog.debug("Depositing %s (%s^3) particles into %s Octs",
             positions.shape[0], positions.shape[0]**0.3333333, nvals[-1])
-        pos = np.asarray(positions.convert_to_units("code_length"),
-                         dtype="float64")
+        positions.convert_to_units("code_length")
```

Comment:
I think as long as `positions` is an internally-generated field, it will be safe to drop the `float64`.  But I did have to stop and think about that.

Reply:
OK, I can add an `astype('float64')`

Reply:
No no, I think `astype` will make a copy, and I think we're safe without it.